### PR TITLE
docs: Minor nits

### PR DESCRIPTION
**Pull request by Billy99**
Fix some minor nits in docs. Primarily, when running as a long running process, use `bpfman system service --timeout=0`. ﻿﻿Also add docker and kind to the list of packages that may need to be installed.

### CODE REVIEW

File: docs/getting-started/building-bpfman.md [hunk 1]
```diff
@@ -241,6 +241,20 @@ sudo dnf install perl
 sudo apt install perl
 ```
 
+### Install docker
+
+To build the `bpfman-agent` and `bpfman-operator`, `docker` needs to be installed.
```
Comment (astoycos) [inline docs/getting-started/building-bpfman.md:246]: I think we need to make sure builds work with podman as well just to be good net citizens :) I thought they did!

Comment (astoycos) [reply docs/getting-started/building-bpfman.md:246]: For KIND we'll always need docker though

Comment (astoycos) [reply docs/getting-started/building-bpfman.md:246]: But at the very least le't just say "container build tooling" or something like that

Comment (Billy99) [reply docs/getting-started/building-bpfman.md:246]: So I agree that should make sure that builds work with podman, but `make build-images` specifically calls `docker build`. I just hit an issue on my server that took an hour to debug. I had `docker` installed but not `docker buildx`, so `make build-images` was failing. So we definitely have a dependency on docker.

**Review by astoycos — APPROVED** [review]
Looks good to me thanks @Billy99!

**Pull request merged**